stargate/crates/query-planner/build.rs

```diff
@@ -0,0 +1,169 @@
+/// and new .feature files.
+fn main() {
+    // If debugging with IJ, use `read_dir("query-planner/tests/features")`
+    // let dirs = read_dir("query-planner/tests/features")
+    let dirs =
+        read_dir(PathBuf::from("tests").join("features"))
+            .expect("read features dir")
+            .map(|res| res.map(|e| e.path()).unwrap())
+            .filter(|d| d.is_dir());
+
+    for dir in dirs {
+        write_composed_schema(&dir).expect("composition");
+        write_tests(&dir).expect("generating test files");
+    }
+}
+
+fn write_composed_schema(dir: &PathBuf) -> std::io::Result<()> {
+    let schema_paths = read_dir(dir)
+        .unwrap()
+        .map(|res| res.map(|e| e.path()).unwrap())
+        .filter(|e| {
+            match (e.extension(), e.file_name()) {
+                (Some(ext), Some(name)) =>
+                    ext == "graphql" && name != "schema.graphql",
+                _ => false,
+            }
+        })
+        .map(|path| {
+            println!("cargo:rerun-if-changed={}", path.to_str().expect("path to string"));
+            path
+        })
+        .map(|path| ServiceDefinition {
+            name: path.file_stem().expect("input schema should have a file stem")
+                .to_str()
+                .expect("file path to string")
+                .to_owned(),
+            type_defs: read_to_string(path)
+                .expect("reading input schema"),
+            url: "undefined".to_owned(),
+        });
+
+    let composed = harmonize(schema_paths.collect());
+    use harmonizer::Result::*;
+    match composed {
+        Ok(schema) => write(dir.join("schema.graphql"), schema),
+        Err(errors) => {
+            let mut message = String::new();
+            for err in &errors {
+                message.push_str(&err.message);
+                message.push('\n');
+            }
+            panic!(message);
+        }
+    }
+}
+
+
+use gherkin_rust::Feature;
+use gherkin_rust::StepType;
+
+
+macro_rules! get_step {
+    ($scenario:ident, $typ:pat) => {
+        $scenario
+            .steps
+            .iter()
+            .find(|s| matches!(s.ty, $typ))
+            .unwrap()
+            .docstring
+            .as_ref()
+            .unwrap()
+    };
+}
+
+fn write_tests(dir: &PathBuf) -> std::io::Result<()> {
+    let feature_paths = read_dir(dir)
+        .unwrap()
+        .map(|res| res.map(|e| e.path()).unwrap())
+        .filter(|e| {
+            if let Some(d) = e.extension() {
+                d == "feature"
+            } else {
+                false
+            }
+        });
+
+    let output_path = dir.with_extension("rs");
+    let mut output = File::create(&output_path)
+        .expect("opening output file");
+
+    let schema_path_str = format!("{}/schema.graphql",
+        dir.file_name().expect("feature dir has filename")
+            .to_str().expect("schema filename"));
+
+    output.write(br#"
+// Autogenerated from build.rs
```

Comment:
This might be obnoxious but can we be a bit louder about this being a generated file? Based on what I mentioned the other day and my manually editing a generated file.

Never got around to landing this since I was trying to eliminate the need for generating files altogether, but i.e.:
https://github.com/apollographql/federation/pull/287/files#diff-983d37a354a6f5a021a1f10fde08e7dd36879fa3e76efe2348698c3276b35b70R17

Reply:
done!